### PR TITLE
Wip: add phpdocs and fix code style

### PR DESCRIPTION
**Pull request by pana1990**
check pr please, it is not completed yet

### CODE REVIEW

**Comment by pana1990** [comment]
I refactor code in SchemaBuilderTrait, check it please

**Comment by pana1990** [comment]
i think to $_dbName should be public In SchemaBuilderTrait...

One question, if i want to use another application id inside migration file how you handle this?

i think to we should readjust migration.php for that it assign the db application component to $dbName.

what do you think about it?

**Comment by vaseninm** [comment]
@pana1990 public or private? both is ok for me. I think, that continue discussion in pull request on origin repository. 
if i want use another component - i use setDb()

**Comment by pana1990** [comment]
@vaseninm ok, i recoment this in origin repository :)

if you use this db option in console  how you set up $_dbName?

**Comment by vaseninm** [comment]
@pana1990 i don't see a problem. 

``` php
Schema::setDb('pgDb')
```

**Comment by pana1990** [comment]
For example, i execute the following conmands console:

```
yii migrate/to m150101_185401_create_news_table --db=db1
yii migrate/to m150101_185401_create_news_table --db=db2
```

How you set up db component in case use?

i think to it should be config in migration.php, for example : 

``` php
// framework/console/controllers/MigrateController.php
public function beforeAction($action)
    {
        if (parent::beforeAction($action)) {
            if ($action->id !== 'create') {
                $this->db = Instance::ensure($this->db, Connection::className());
                Schema::setDb($this->db); // Here set up db component
            }
            return true;
        } else {
            return false;
        }
    }
```

**Comment by vaseninm** [comment]
Oh, thanks. I did not know about it. I do it, but i think, that all technical discussion must be in origin repo. Lets finished phpdocs and i merge it.

**Comment by pana1990** [comment]
it is truth, i recomment this question.

Thanks for your job ;)

**Comment by vaseninm** [comment]
@pana1990 thanks for phpdocs. It's very important job.

**Comment by pana1990** [comment]
you are welcome ;)

**Comment by vaseninm** [comment]
@pana1990 thanks. Phpdocs are not finished now?

**Comment by pana1990** [comment]
The public methods and class files finished but it should be completed by samdark or quiang... you may merge for that they checks changes

**Comment by vaseninm** [comment]
But yii has description of non-public methods. And magic method of SchemaBuilder has not documentation.

**Comment by pana1990** [comment]
it is turh, done!

I think it should be improved documentation by the core developers but this help them.

Pd: if Refactor code is merged i should add @method comments for support to autocomplete IDE.

**Comment by vaseninm** [comment]
Good. Add @methods, plz, and i do final review and merge it.

**Comment by pana1990** [comment]
@vaseninm ok ;)

**Comment by pana1990** [comment]
@vaseninm  ready!

File: framework/db/SchemaBuilder.php [hunk 1]
```diff
@@ -14,101 +11,242 @@
...
+     * @return SchemaBuilder
+     */
     public static function decimal($precision = null, $scale = null)
     {
         return static::createNumeric(Schema::TYPE_DECIMAL, $precision, $scale);
     }
 
+    /**
+     * Specify type of field as datetime
+     *
```
Comment (vaseninm) [inline framework/db/SchemaBuilder.php:173]: @param skipped

File: framework/db/SchemaBuilder.php [hunk 2]
```diff
@@ -14,101 +11,242 @@
...
+     * @return SchemaBuilder
+     */
     public static function dateTime($length = null)
     {
         return static::createDefault(Schema::TYPE_DATETIME, $length);
     }
 
+    /**
+     * Specify type of field as timestamp
+     *
```
Comment (vaseninm) [inline framework/db/SchemaBuilder.php:183]: @param skipped

File: framework/db/SchemaBuilder.php [hunk 3]
```diff
@@ -14,101 +11,242 @@
...
+     * @return SchemaBuilder
+     */
     public static function timestamp($length = null)
     {
         return static::createDefault(Schema::TYPE_TIMESTAMP, $length);
     }
 
+    /**
+     * Specify type of field as time
+     *
```
Comment (vaseninm) [inline framework/db/SchemaBuilder.php:193]: @param skipped

File: framework/db/SchemaBuilder.php [hunk 4]
```diff
@@ -14,101 +11,242 @@
...
+     * @return SchemaBuilder
+     */
     public static function time($length = null)
     {
         return static::createDefault(Schema::TYPE_TIME, $length);
     }
 
+    /**
+     * Specify type of field as date
+     *
```
Comment (vaseninm) [inline framework/db/SchemaBuilder.php:203]: @param skipped

File: framework/db/SchemaBuilder.php [hunk 5]
```diff
@@ -178,11 +359,21 @@ protected function getDefaultString()
...
+     * Returns check value of column
+     *
+     * @return string
+     */
     protected function getCheckString()
     {
         return ($this->check !== null ? " CHECK ({$this->check})" : '');
     }
 
+    /**
```
Comment (vaseninm) [inline framework/db/SchemaBuilder.php:372]: description skipped

File: framework/db/SchemaBuilder.php [hunk 6]
```diff
@@ -193,6 +384,12 @@ protected static function createDefault($type, $length = null)
         return $object;
     }
 
+    /**
+     * @param string $type schema of column
```
Comment (vaseninm) [inline framework/db/SchemaBuilder.php:392]: description skipped

**Comment by vaseninm** [comment]
done.

**Comment by pana1990** [comment]
the following methods should be without params : 
- dateTime()
- timestamp()
- time()
- date()

what do you think?

some idea for description createDefault and createNumeric methods?

**Comment by vaseninm** [comment]
Yes. dateTime and others should be without params. I'll fix it tomorrow.

Create schema builder for types with length
Create schema builder for numeric types types with precision and scale.

**Comment by pana1990** [comment]
done! :)

**Comment by vaseninm** [comment]
Thank. Good job. Merged.

**Pull request merged**